workflow-ui/core-android/src/main/java/com/squareup/workflow1/ui/WorkflowRunnerViewModel.kt

```diff
@@ -74,26 +74,27 @@ internal class WorkflowRunnerViewModel<OutputT>(
       val config = configure()
       val props = config.props
       val scope = CoroutineScope(config.dispatcher)
-      val result = CompletableDeferred<OutputT>(parent = scope.coroutineContext[Job])
+      val outputChannel = Channel<OutputT>()
+      scope.coroutineContext[Job]!!.invokeOnCompletion {
+        outputChannel.close(it)
```

Comment:
I think we need to check if `it` is a `CancellationException` and pass null through if it is. I think you said there's test coverage for this though, so I guess it works?

Reply:
These are the test scenarios AFAICT:
1. Verify cancellation is propagated to running workers when the runner's scope is cleared. `Factory cancels runtime when cleared`
2. Verify that the coroutine that is suspend-waiting on the next output is canceled when the runner's scope is cleared. `Factory cancels result when cleared`
3. When the runner scope is canceled, verify that a CancellationException is invoked on completion of runnerScope.coroutineContext[Job], and that the cause-chain of this exception are only CancellationExceptions. `hostCancelledOnCleared`

Is there another scenario we should test?

>check if it is a CancellationException and pass null through if it is
So we want future usages of the channel to throw a ClosedSendChannelException/ClosedReceiveChannelException instead of a CancellationException? Or is there another reason to do this?